util/pom.xml

```diff
@@ -74,6 +74,13 @@
             <artifactId>mail</artifactId>
             <scope>test</scope>
         </dependency>
+        <!-- nullability -->
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
```

Comment:
Do we really need this dependency in util for null check ?

Reply:
In general I prefer to annotate parameters and return values but the dependency is of course not strictly required. The annotations (https://github.com/JetBrains/java-annotations) are used during analysis and compile time only and should have no negative impact.
Should I remove the annotations from ConfigurationPID?

Reply:
Don't worry, I'm reworking your two PRs. Thanks !